datahub-graphql-core/src/test/java/com/linkedin/datahub/graphql/resolvers/role/BatchAssignRoleToActorsResolverTest.java

```diff
@@ -0,0 +1,106 @@
+    when(_dataFetchingEnvironment.getContext()).thenReturn(mockContext);
+    when(mockContext.getAuthentication()).thenReturn(_authentication);
+
+    BatchAssignRoleToActorsInput input = new BatchAssignRoleToActorsInput();
+    input.setRoleUrn(ROLE_URN_STRING);
+    List<String> actors = new ArrayList<>();
+    actors.add(FIRST_ACTOR_URN_STRING);
+    input.setActors(actors);
+    when(_dataFetchingEnvironment.getArgument("input")).thenReturn(input);
+    when(_entityClient.exists(eq(Urn.createFromString(ROLE_URN_STRING)), any())).thenReturn(false);
+
+    assertThrows(() -> _resolver.get(_dataFetchingEnvironment).join());
+  }
+
+  @Test
+  public void testSomeActorsExist() throws Exception {
+    QueryContext mockContext = getMockAllowContext();
+    when(_dataFetchingEnvironment.getContext()).thenReturn(mockContext);
+    when(mockContext.getAuthentication()).thenReturn(_authentication);
+
+    BatchAssignRoleToActorsInput input = new BatchAssignRoleToActorsInput();
+    input.setRoleUrn(ROLE_URN_STRING);
+    List<String> actors = new ArrayList<>();
+    actors.add(FIRST_ACTOR_URN_STRING);
+    actors.add(SECOND_ACTOR_URN_STRING);
+    input.setActors(actors);
+    when(_dataFetchingEnvironment.getArgument(eq("input"))).thenReturn(input);
+    when(_entityClient.exists(eq(Urn.createFromString(ROLE_URN_STRING)), any())).thenReturn(true);
+    when(_entityClient.exists(eq(Urn.createFromString(FIRST_ACTOR_URN_STRING)), any())).thenReturn(true);
+    when(_entityClient.exists(eq(Urn.createFromString(SECOND_ACTOR_URN_STRING)), any())).thenReturn(false);
+
+    assertTrue(_resolver.get(_dataFetchingEnvironment).join());
+    // Only the first actor should be assigned to the role since the second actor does not exist
+    verify(_entityClient, times(1)).ingestProposal(any(), any());
+  }
+
+  @Test
+  public void testAllActorsExist() throws Exception {
+    QueryContext mockContext = getMockAllowContext();
+    when(_dataFetchingEnvironment.getContext()).thenReturn(mockContext);
+    when(mockContext.getAuthentication()).thenReturn(_authentication);
+
+    BatchAssignRoleToActorsInput input = new BatchAssignRoleToActorsInput();
+    input.setRoleUrn(ROLE_URN_STRING);
+    List<String> actors = new ArrayList<>();
+    actors.add(FIRST_ACTOR_URN_STRING);
+    actors.add(SECOND_ACTOR_URN_STRING);
+    input.setActors(actors);
+    when(_dataFetchingEnvironment.getArgument(eq("input"))).thenReturn(input);
+    when(_entityClient.exists(eq(Urn.createFromString(ROLE_URN_STRING)), any())).thenReturn(true);
+    when(_entityClient.exists(eq(Urn.createFromString(FIRST_ACTOR_URN_STRING)), any())).thenReturn(true);
+    when(_entityClient.exists(eq(Urn.createFromString(SECOND_ACTOR_URN_STRING)), any())).thenReturn(true);
+
+    assertTrue(_resolver.get(_dataFetchingEnvironment).join());
+    // Both actors exist and should be assigned to the role
+    verify(_entityClient, times(2)).ingestProposal(any(), any());
```

Comment:
Minor: can we do better than "any()" comparisons here?

Reply:
Added checks for authentication, the proposal might be a bit harder...